[CI] CrossClusterAsyncQueryStopIT testStopQueryInlineStats failing

**Build Scans:**
- [elasticsearch-intake #28680 / part3](https://gradle-enterprise.elastic.co/s/xqiz67swuq2bs)
- [elasticsearch-pull-request #93378 / part-3](https://gradle-enterprise.elastic.co/s/lqxmof6nlgepo)
- [elasticsearch-intake #28676 / part3](https://gradle-enterprise.elastic.co/s/g7ofm34xfn3ze)
- [elasticsearch-pull-request #93373 / part-3](https://gradle-enterprise.elastic.co/s/pdaosud6naryu)
- [elasticsearch-pull-request #93361 / part-3](https://gradle-enterprise.elastic.co/s/cm7zn5keb5bhy)
- [elasticsearch-pull-request #93363 / part-3](https://gradle-enterprise.elastic.co/s/lkvvb4jdbzega)
- [elasticsearch-pull-request #93361 / part-3](https://gradle-enterprise.elastic.co/s/v7gqkswbgsrli)
- [elasticsearch-pull-request #93358 / part-3](https://gradle-enterprise.elastic.co/s/sd45tsrndcn5a)
- [elasticsearch-pull-request #93359 / part-3](https://gradle-enterprise.elastic.co/s/dwyftrd6zqaws)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.CrossClusterAsyncQueryStopIT.testStopQueryInlineStats" -Dtests.seed=6E5FDC10FB8B212B -Dtests.locale=kxv-Telu-IN -Dtests.timezone=America/St_Kitts -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.CrossClusterAsyncQueryStopIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testStopQueryInlineStats'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: null
     but: was <45L>
```

**Issue Reasons:**
- [main] 9 failures in test testStopQueryInlineStats (33.3% fail rate in 27 executions)
- [main] 2 failures in step part3 (22.2% fail rate in 9 executions)
- [main] 7 failures in step part-3 (38.9% fail rate in 18 executions)
- [main] 2 failures in pipeline elasticsearch-intake (22.2% fail rate in 9 executions)
- [main] 6 failures in pipeline elasticsearch-pull-request (37.5% fail rate in 16 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -510,9 +510,6 @@ tests:
 - class: org.elasticsearch.repositories.blobstore.testkit.analyze.MinioRepositoryAnalysisRestIT
   method: testRepositoryAnalysis
   issue: https://github.com/elastic/elasticsearch/issues/134853
-- class: org.elasticsearch.xpack.esql.action.CrossClusterAsyncQueryStopIT
-  method: testStopQueryInlinestats
-  issue: https://github.com/elastic/elasticsearch/issues/134854
 - class: org.elasticsearch.xpack.esql.action.CrossClusterCancellationIT
   method: testCancelSkipUnavailable
   issue: https://github.com/elastic/elasticsearch/issues/134865
@@ -528,9 +525,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
   method: test {csv-spec:stats.CountDistinctWithConditions}
   issue: https://github.com/elastic/elasticsearch/issues/134993
-- class: org.elasticsearch.xpack.esql.action.CrossClusterAsyncQueryStopIT
-  method: testStopQueryInlineStats
-  issue: https://github.com/elastic/elasticsearch/issues/135032
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
   method: test {csv-spec:fork.ForkBeforeStatsWithWhere}
   issue: https://github.com/elastic/elasticsearch/issues/135041

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterAsyncQueryStopIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterAsyncQueryStopIT.java
@@ -314,9 +314,13 @@ public class CrossClusterAsyncQueryStopIT extends AbstractCrossClusterTestCase {
                 assertThat(asyncResponse.columns().size(), equalTo(2));
                 AtomicInteger i = new AtomicInteger(0);
                 asyncResponse.values().forEachRemaining(row -> {
-                    // We will have all rows here but total will be null since we stopped the inline stats before it could complete
-                    assertThat(row.next(), equalTo(null));
                     var v = row.next();
+                    // The sum could be null, if the stats did not manage to compute anything before being stopped
+                    // Or it could be 45L if it managed to add 0-9
+                    if (v != null) {
+                        assertThat((long) v, equalTo(45L));
+                    }
+                    v = row.next();
                     if (v != null) {
                         assertThat((long) v, lessThanOrEqualTo(10L));
                     }

